src/Tool.tsx

```diff
@@ -73,36 +73,37 @@ export const DarkMode: React.FunctionComponent<DarkModeProps> = props => {
       ...currentStore,
       current
     });
-
     props.api.setOptions({ theme: currentStore[current] });
     setDark(!isDark);
     props.channel.emit('DARK_MODE', !isDark);
   }
 
-  React.useEffect(() => {
-    const channel = props.api.getChannel();
+  function renderTheme() {
+    const { parameters } = props.api.getCurrentStoryData();
 
-    channel.on('storiesConfigured', () => {
-      const { parameters } = props.api.getCurrentStoryData();
+    let darkTheme = themes.dark;
+    let lightTheme = themes.light;
 
-      let darkTheme = themes.dark;
-      let lightTheme = themes.light;
+    if (parameters && parameters.darkMode) {
+      darkTheme = parameters.darkMode.dark || darkTheme;
+      lightTheme = parameters.darkMode.light || lightTheme;
+    }
 
-      if (parameters && parameters.darkMode) {
-        darkTheme = parameters.darkMode.dark || darkTheme;
-        lightTheme = parameters.darkMode.light || lightTheme;
-      }
+    const currentStore = store({
+      light: lightTheme,
+      dark: darkTheme
+    });
+    const { current } = currentStore;
 
-      const currentStore = store({
-        light: lightTheme,
-        dark: darkTheme
-      });
-      const { current } = currentStore;
+    props.api.setOptions({ theme: currentStore[current] });
+    setDark(current === 'dark');
+    props.channel.emit('DARK_MODE', current === 'dark');
+  }
 
-      props.api.setOptions({ theme: currentStore[current] });
-      setDark(current === 'dark');
-      props.channel.emit('DARK_MODE', current === 'dark');
-    });
+  React.useEffect(() => {
+    const channel = props.api.getChannel();
+    channel.on('storyChanged', () => renderTheme());
```

Comment:
Could this just be the following?

```suggestion
    channel.on('storyChanged', renderTheme);
```

Reply:
I think that there not be any problem! I'll commit the change

Reply:
Changes made @hipstersmoothie !